src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -44,15 +50,29 @@ public ProductResponse saveProduct(CreateProductRequest createProductRequest, Mu
     }
 
     @Override
-    public Product findProductByIdOrThrowException(UUID productId) {
-        return productRepository.findById(productId)
-                .orElseThrow(() -> new NoSuchElementException("Product with id " + productId + " does not exist."));
+    public ProductResponse getProduct(UUID productId) {
+        Product product = findProductByIdOrThrowException(productId);
+        return ProductMapper.toProductResponse(product);
+    }
```

Comment:
Acá nos podemos evitar guardar en memoria un producto si combinamos los métodos

```suggestion
    public ProductResponse getProduct(UUID productId) {
        return ProductMapper.toProductResponse(
            findProductByIdOrThrowException(productId)
        );
    }
```